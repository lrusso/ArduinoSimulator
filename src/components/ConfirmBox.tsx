import React from "react"

interface ConfirmBoxProps {
  title: string
  message: string
  accept: string
  acceptCallback: () => void
  cancel: string
  cancelCallback: () => void
}

const ConfirmBox = ({
  title,
  message,
  accept,
  acceptCallback,
  cancel,
  cancelCallback,
}: ConfirmBoxProps) => {
  return (
    <div style={styles.container}>
      <div style={styles.form}>
        <div style={styles.title}>{title}</div>
        <div style={styles.message}>{message}</div>
        <div style={styles.buttonContainer}>
          <input
            type="button"
            style={styles.button}
            onClick={acceptCallback}
            value={accept}
          />
          <div style={styles.spacing}></div>
          <input
            type="button"
            style={styles.button}
            onClick={cancelCallback}
            value={cancel}
          />
        </div>
      </div>
    </div>
  )
}

const styles: { [key: string]: React.CSSProperties } = {
  container: {
    position: "fixed",
    minWidth: "320px",
    top: "0",
    left: "0",
    right: "0",
    bottom: "0",
    zIndex: "8888",
    backgroundColor: "rgba(0,0,0,0.5)",
    userSelect: "none",
    WebkitUserSelect: "none",
  },
  form: {
    position: "relative",
    top: "50%",
    left: "0",
    right: "0",
    marginLeft: "auto",
    marginRight: "auto",
    transform: "translateY(-50%)",
    width: "300px",
    backgroundColor: "#181818",
    textAlign: "center",
    borderRadius: "5px",
    overflow: "hidden",
    boxShadow: "0 2px 2px 1px rgba(0,0,0,0.4)",
  },
  title: {
    textAlign: "left",
    paddingLeft: "10px",
    backgroundColor: "#3a76b1",
    fontFamily: "'Segoe UI', Tahoma, Geneva, Verdana, sans-serif",
    fontSize: "16px",
    fontWeight: "bold",
    color: "white",
    lineHeight: "2.5",
    cursor: "default",
  },
  message: {
    marginLeft: "10px",
    marginRight: "10px",
    paddingTop: "20px",
    paddingBottom: "10px",
    fontFamily: "'Segoe UI', Tahoma, Geneva, Verdana, sans-serif",
    fontSize: "16px",
    fontWeight: "400",
    color: "black",
    overflow: "hidden",
    textAlign: "center",
    lineHeight: "2",
    cursor: "default",
    whiteSpace: "pre-wrap",
  },
  buttonContainer: {
    float: "left",
    width: "300px",
    textAlign: "center",
  },
  button: {
    outline: "none",
    padding: "10px",
    backgroundColor: "#d2d2d2",
    border: "1px solid #b2b2b2",
    fontFamily: "'Segoe UI', Tahoma, Geneva, Verdana, sans-serif",
    fontSize: "16px",
    color: "black",
    textAlign: "center",
    lineHeight: "1.5",
    display: "inline-block",
    marginBottom: "10px",
    paddingLeft: "40px",
    paddingRight: "40px",
    cursor: "pointer",
    borderRadius: "5px",
  },
  spacing: {
    display: "inline-block",
    width: "20px",
    height: "40px",
  },
}

export default ConfirmBox
