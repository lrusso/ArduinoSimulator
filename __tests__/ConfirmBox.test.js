import React from "react"
import renderer from "react-test-renderer"
import ConfirmBox from "../src/components/ConfirmBox"

describe("ConfirmBox", () => {
  test("Rendering 1", () => {
    const tree = renderer
      .create(
        <ConfirmBox
          title="Test Title"
          message="Test Message"
          accept={"Test Accept"}
          acceptCallback={() => {}}
          cancel={"Test Cancel"}
          cancelCallback={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
