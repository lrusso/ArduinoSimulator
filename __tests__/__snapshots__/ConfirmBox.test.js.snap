// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmBox Rendering 1 1`] = `
<div
  style={
    {
      "WebkitUserSelect": "none",
      "backgroundColor": "rgba(0,0,0,0.5)",
      "bottom": "0",
      "left": "0",
      "minWidth": "320px",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "userSelect": "none",
      "zIndex": "8888",
    }
  }
>
  <div
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": "5px",
        "boxShadow": "0 2px 2px 1px rgba(0,0,0,0.4)",
        "left": "0",
        "marginLeft": "auto",
        "marginRight": "auto",
        "overflow": "hidden",
        "position": "relative",
        "right": "0",
        "textAlign": "center",
        "top": "50%",
        "transform": "translateY(-50%)",
        "width": "300px",
      }
    }
  >
    <div
      style={
        {
          "backgroundColor": "#3a76b1",
          "color": "white",
          "cursor": "default",
          "fontFamily": "Arial",
          "fontSize": "16px",
          "fontWeight": "bold",
          "lineHeight": "2.5",
          "paddingLeft": "10px",
          "textAlign": "left",
        }
      }
    >
      Test Title
    </div>
    <div
      style={
        {
          "color": "black",
          "cursor": "default",
          "fontFamily": "Arial",
          "fontSize": "16px",
          "fontWeight": "400",
          "lineHeight": "2",
          "marginLeft": "10px",
          "marginRight": "10px",
          "overflow": "hidden",
          "paddingBottom": "10px",
          "paddingTop": "20px",
          "textAlign": "center",
          "whiteSpace": "pre-wrap",
        }
      }
    >
      Test Message
    </div>
    <div
      style={
        {
          "float": "left",
          "textAlign": "center",
          "width": "300px",
        }
      }
    >
      <input
        onClick={[Function]}
        style={
          {
            "backgroundColor": "#d2d2d2",
            "border": "1px solid #b2b2b2",
            "borderRadius": "5px",
            "color": "black",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "lineHeight": "1.5",
            "marginBottom": "10px",
            "outline": "none",
            "padding": "10px",
            "paddingLeft": "40px",
            "paddingRight": "40px",
            "textAlign": "center",
          }
        }
        type="button"
        value="Test Accept"
      />
      <div
        style={
          {
            "display": "inline-block",
            "height": "40px",
            "width": "20px",
          }
        }
      />
      <input
        onClick={[Function]}
        style={
          {
            "backgroundColor": "#d2d2d2",
            "border": "1px solid #b2b2b2",
            "borderRadius": "5px",
            "color": "black",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Arial",
            "fontSize": "16px",
            "lineHeight": "1.5",
            "marginBottom": "10px",
            "outline": "none",
            "padding": "10px",
            "paddingLeft": "40px",
            "paddingRight": "40px",
            "textAlign": "center",
          }
        }
        type="button"
        value="Test Cancel"
      />
    </div>
  </div>
</div>
`;
